ethcore/types/src/basic_account.rs

```diff
@@ -29,4 +29,48 @@ pub struct BasicAccount {
 	pub storage_root: H256,
 	/// Code hash of the account.
 	pub code_hash: H256,
+	/// Code version of the account.
+	pub code_version: U256,
+}
+
+impl rlp::Encodable for BasicAccount {
+	fn rlp_append(&self, stream: &mut rlp::RlpStream) {
+		let use_short_version = self.code_version == U256::zero();
+
+		match use_short_version {
+			true => { stream.begin_list(4); }
+			false => { stream.begin_list(5); }
+		}
+
+		stream.append(&self.nonce);
+		stream.append(&self.balance);
+		stream.append(&self.storage_root);
+		stream.append(&self.code_hash);
+
+		if !use_short_version {
+			stream.append(&self.code_version);
+		}
+	}
+}
+
+impl rlp::Decodable for BasicAccount {
+	fn decode(rlp: &rlp::Rlp) -> Result<Self, rlp::DecoderError> {
+		let use_short_version = match rlp.item_count()? {
```

Comment:
The same comment about fragile logic based on item count

Reply:
Using `rlp.item_count()` is the only method to know whether we're at version `0` or above version `0`. I don't agree it's fragile -- `BasicAccount` is guaranteed to be either have 4 items (version `0` case) or 5 items (version non-`0` case). And we handle all errors explicitly.